packages/st2/rpm/st2.spec

```diff
@@ -25,6 +25,24 @@ BuildRequires: python3-devel
 BuildRequires: python3-setuptools
 %endif  # Requires for RHEL 8
 
+%if 0%{?rhel} >= 7
+# By default on EL 8, RPM helper scripts will try to generate Requires: section which lists every
+# Python dependencies. That process / script works by recursively scanning all the package Python
+# dependencies which is very slow (5-6 minutes).
+# Our package bundles virtualenv with all the dependendencies and doesn't rely on this metadata
+# so we skip that step to vastly speed up the build.
+# Technically we also don't Require or Provide any of those libraries auto-detected by that script
+# because those are only used internally inside a package specific virtual environment.
+# Same step also does not run on EL7.
+# See https://github.com/StackStorm/st2-packages/pull/697#issuecomment-808971874 and that PR for
+# more details.
+# That issue was found by enabling rpmbuild -vv flag.
```

Comment:
Can you show an example of `Requires:` record before and after this change?

There might be important dependencies in there like shared libraries that are indeed required for the package to work properly.

Reply:
Here is how `Requires` currently looks like:
```

Requires

    /bin/bash
    /bin/sh
    /bin/sh
    /opt/stackstorm/st2/bin/python
    /usr/bin/env
    bash
    config(st2) = 3.4.1-2
    git
    ld-linux-x86-64.so.2()(64bit)
    ld-linux-x86-64.so.2(GLIBC_2.3)(64bit)
    libc.so.6(GLIBC_2.14)(64bit)
    libcrypto.so.1.1()(64bit)
    libdl.so.2()(64bit)
    libffi-806b1a9d.so.6.0.4()(64bit)
    libffi-devel
    libgcc_s.so.1()(64bit)
    liblber-2.4.so.2()(64bit)
    libldap_r-2.4.so.2()(64bit)
    libm.so.6()(64bit)
    libpthread.so.0()(64bit)
    libpthread.so.0(GLIBC_2.2.5)(64bit)
    libpython3.6m.so.1.0()(64bit)
    libstdc++.so.6()(64bit)
    libutil.so.1()(64bit)
    libyaml-0.so.2()(64bit)
    openssh-clients
    openssh-server
    openssl-devel
    pam
    python3-devel
    rtld(GNU_HASH)
    setup

```
https://packagecloud.io/StackStorm/stable/packages/el/8/st2-3.4.1-2.x86_64.rpm

Reply:
Yeah, see the comment here - https://github.com/StackStorm/st2-packages/pull/697#issuecomment-808971874 (for ``Provides``) and the output below.

Requires EL 8 before:

```bash
rpm -iqp --requires ~/Downloads/st2-3.5dev-28.x86_64.rpm
/bin/bash
/bin/sh
/bin/sh
/bin/sh
/bin/sh
/bin/sh
/opt/stackstorm/st2/bin/python
/usr/bin/env
bash
config(st2) = 3.5dev-28
git
ld-linux-x86-64.so.2()(64bit)
ld-linux-x86-64.so.2(GLIBC_2.3)(64bit)
libc.so.6()(64bit)
libc.so.6(GLIBC_2.13)(64bit)
libc.so.6(GLIBC_2.14)(64bit)
libc.so.6(GLIBC_2.2.5)(64bit)
libc.so.6(GLIBC_2.3)(64bit)
libc.so.6(GLIBC_2.3.2)(64bit)
libc.so.6(GLIBC_2.3.4)(64bit)
libc.so.6(GLIBC_2.4)(64bit)
libc.so.6(GLIBC_2.6)(64bit)
libc.so.6(GLIBC_2.7)(64bit)
libcrypto.so.1.1()(64bit)
libdl.so.2()(64bit)
libffi-806b1a9d.so.6.0.4()(64bit)
libffi-devel
libgcc_s.so.1()(64bit)
libgcc_s.so.1(GCC_3.0)(64bit)
liblber-2.4.so.2()(64bit)
libldap_r-2.4.so.2()(64bit)
libm.so.6()(64bit)
libm.so.6(GLIBC_2.2.5)(64bit)
libpthread.so.0()(64bit)
libpthread.so.0(GLIBC_2.2.5)(64bit)
libpython3.6m.so.1.0()(64bit)
libstdc++.so.6()(64bit)
libstdc++.so.6(CXXABI_1.3)(64bit)
libstdc++.so.6(GLIBCXX_3.4)(64bit)
libutil.so.1()(64bit)
openssh-clients
openssh-server
openssl-devel
pam
python3-devel
rpmlib(CompressedFileNames) <= 3.0.4-1
rpmlib(FileDigests) <= 4.6.0-1
rpmlib(PayloadFilesHavePrefix) <= 4.0-1
rpmlib(PayloadIsXz) <= 5.2-1
rtld(GNU_HASH)
setup
```

Requires EL 8 after:


```bash
kami ~/w/stackstorm/st2-packages (git:cache_tmp_wheelhouse)$ rpm -iqp --requires ~/Downloads/st2-3.5dev-28.x86_64*2*.rpm
/bin/bash
/bin/sh
/bin/sh
/bin/sh
/bin/sh
/bin/sh
/opt/stackstorm/st2/bin/python
/usr/bin/env
bash
config(st2) = 3.5dev-28
git
ld-linux-x86-64.so.2()(64bit)
ld-linux-x86-64.so.2(GLIBC_2.3)(64bit)
libc.so.6()(64bit)
libc.so.6(GLIBC_2.13)(64bit)
libc.so.6(GLIBC_2.14)(64bit)
libc.so.6(GLIBC_2.2.5)(64bit)
libc.so.6(GLIBC_2.3)(64bit)
libc.so.6(GLIBC_2.3.2)(64bit)
libc.so.6(GLIBC_2.3.4)(64bit)
libc.so.6(GLIBC_2.4)(64bit)
libc.so.6(GLIBC_2.6)(64bit)
libc.so.6(GLIBC_2.7)(64bit)
libcrypto.so.1.1()(64bit)
libdl.so.2()(64bit)
libffi-806b1a9d.so.6.0.4()(64bit)
libffi-devel
libgcc_s.so.1()(64bit)
libgcc_s.so.1(GCC_3.0)(64bit)
liblber-2.4.so.2()(64bit)
libldap_r-2.4.so.2()(64bit)
libm.so.6()(64bit)
libm.so.6(GLIBC_2.2.5)(64bit)
libpthread.so.0()(64bit)
libpthread.so.0(GLIBC_2.2.5)(64bit)
libpython3.6m.so.1.0()(64bit)
libstdc++.so.6()(64bit)
libstdc++.so.6(CXXABI_1.3)(64bit)
libstdc++.so.6(GLIBCXX_3.4)(64bit)
libutil.so.1()(64bit)
openssh-clients
openssh-server
openssl-devel
pam
python3-devel
rpmlib(CompressedFileNames) <= 3.0.4-1
rpmlib(FileDigests) <= 4.6.0-1
rpmlib(PayloadFilesHavePrefix) <= 4.0-1
rpmlib(PayloadIsXz) <= 5.2-1
rtld(GNU_HASH)
setup
```

In short, that change only affect ``Provides`` for Python packages and ``Requires`` itself is unchanged.

Reply:
@armab 

> As for Provides, there's a lot of unneeded noise there, if a cleanup helps in a build speedup that's great indeed.

As per my comment here (https://github.com/StackStorm/st2-packages/pull/697#issuecomment-808974651), I don't think it's just noise.

It's actually incorrect and broken since ``Provides`` claims that package provides a bunch of other Python libraries which is not correct - those libraries are not provided by the package and are just used internally in the virtual environment.

Likely no one was affected by this bug yet since package is usually installed on isolated VMs (and as per comment above, to be "affected by it" would require some specific steps / dependencies to be installed).

Reply:
If you believe how the `Provides` was generated is a bug in the package building, do we need to also include the same change for any RPM package, not just EL8?

Reply:
Per the comment - the change only affects EL 8 and not EL 7.

I didn't dig in that deep to figure out exactly why, but I assume it's due to the differences how that script works which automatically finds all the transitive Python package dependencies and declares them under ``Provides``.

Having said that, I also added a change to use that section on EL 7 as well for just in case.

Reply:
That makes sense, indeed it's all good for EL7: https://packagecloud.io/StackStorm/stable/packages/el/7/st2-3.4.1-2.x86_64.rpm